sympy/crypto/crypto.py

```diff
@@ -3324,3 +3325,138 @@ def decipher_bg(message, key):
         orig_msg += (m ^ b)
 
     return orig_msg
+
+################# Columnar Transposition Cipher #################
+
+def encipher_columnar(message, key):
+    """
+    Performs Regular Columnar Encryption on plaintext and returns ciphertext
+
+    Examples
+    ========
+
+    >> from sympy.crypto.crypto import encipher_columnar
+    >> message = "HELLO WORLD"
+    >> encipher_columnar(message,"SYM")
+    "LWLEHLODEORY"
+
+    Parameters
+    ==========
+
+    message : string, the message to be encrypted.
+    key : string, the key to be used for encryption.
+
+    Returns
+    =======
+
+    The Encrypted message.
+
+    References
+    ==========
+    https://en.wikipedia.org/wiki/Transposition_cipher#Columnar_transposition
+
+    """
+    message =  ''.join(alphabet for alphabet in message if alphabet.isalpha())
+    key =  ''.join(alphabet for alphabet in key if alphabet.isalpha())
+    row_length = len(key)
+    words = len(message)
+    col_length = -(-words//row_length)          # Ceil Function
+    remainder = words%row_length
+
+    for i in range(row_length - remainder):
+        message = message + str(chr(randint(0,25)+97))      # Padding message for regular columnar transposition
+
+    rows = [ord(word) for word in key.lower()]
+    temp = sorted(rows)
+    index_list = [temp.index(v)+1 for v in rows]        # List of ranks of words of key
+    matrix = [message[i:i+row_length] for i in range(0,words,row_length)]
+
+    # Modification for keys with repetitive words
+    if len(index_list) != len(set(index_list)):
+        curr_index = -1
+        add = 0
+        for i in range(row_length):
+            if index_list.count(i) > 1:
+                while index_list.count(i) > 1:
+                    curr_index = index_list.index(i,curr_index + 1)
+                    index_list[curr_index] += add
+                    add += 1
+            curr_index = -1
+            add = 0
+
+    cipher = ""
```

Comment:
I think you should use a list here `cipher = []` into which letters are added then return `''.join(cipher)` at the end.

Reply:
I have pushed the changes you suggested. I didn't know the difference list makes in terms of storage as compared to a string. Thanks for the review.